ui/src/shared/components/HeatmapPlot.tsx

```diff
@@ -85,6 +89,9 @@ const HeatmapPlot: FunctionComponent<Props> = ({
     timeZone,
   })
 
+  xDomain[0] = Math.min(startTime, xDomain[0])
+  xDomain[1] = Math.max(endTime, xDomain[1])
```

Comment:
i noticed that `xDomain` is coming from `useVisDomainSettings` which seems to be used in a lot of plots. Do all of those plots need the same treatment, or just `ScatterPlot` and `HeatmapPlot`?

i think it's probably clearer and less error prone and easier to reason about to assign these values to a new variable, rather than reassigning them to `xDomain`.  the compiler would complain about this:

```ts
const singleNumberValue = useOtherDomainSettings()
singleNumbervalue = Math.max(someOtherDamnVariable, singleNumberValue)
```

```suggestion
  const remappedXDomain = [Math.min(startTime, xDomain[0]), Math.max(endTime, xDomain[1])]
```